joomla/joomla-cms
# [4.0] Media field Image select Features parity

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/31319 .

### Summary of Changes
Should solve the release blocker https://github.com/joomla/joomla-cms/issues/31319

- It restores the ability to create a Figure element around the image. This happens only if you provide some figure caption text

- It restores the ability to insert custom classes to an image (space separated)

- It also introduces the `No Description` checkbox that @brianteeman introduced in other places (so basically now it's also available for the images inserted into an editor)

But also

- It doesn't bring the `Position` field to Joomla 4. The reason is that this has already been converted in the other places to use classes, ref: https://github.com/joomla/joomla-cms/pull/31017
- It doesn't bing also title to the available image tags. The reason is that title adds nothing to a11y, but don't take my word for this, read about title in MDN: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img (search for `title attribute`)

![Screenshot 2020-11-18 at 16 13 09](https://user-images.githubusercontent.com/3889375/99551316-fae70780-29bb-11eb-9ee2-b7bc9e3d7ef8.png)
![Screenshot 2020-11-18 at 16 10 17](https://user-images.githubusercontent.com/3889375/99551330-fe7a8e80-29bb-11eb-9096-63c75d62e1e6.png)
![Screenshot 2020-11-18 at 16 12 45](https://user-images.githubusercontent.com/3889375/99551339-020e1580-29bc-11eb-89b4-7aba20f589bc.png)

#### Styling

Before going nuclear on this, let me just say that I introduced the functional part of the code. I just styled (adding a position and a background) so it can be tested. Feel free to style it as you like, the css lives @ `build/media_source/system/css/fields/joomla-image-select.css`. Also please don't ask me to style it, I'm not a fan of bootstrap

### Testing Instructions

This PR needs rebuilding of the assets so either `npm I` or download the package for the PR!!!

Edit an article.
In the editor insert an image using the xtd-button.
Try all the combos and check the inserted text (just toggle the editor)


### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Documentation Changes Required
No (?)
